packages/node/src/edge-functions/edge-handler-template.js

```diff
@@ -62,15 +62,31 @@ async function parseRequestEvent(event) {
 
 // This will be invoked by logic using this template
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-function registerFetchListener(userEdgeHandler, options, dependencies) {
+function registerFetchListener(module, options, dependencies) {
   addEventListener('fetch', async event => {
+    let handler;
+    if (typeof module.default === 'function') {
+      handler = module.default;
+    } else {
+      if (
+        ['GET', 'HEAD', 'OPTIONS', 'POST', 'PUT', 'DELETE', 'PATCH'].some(
+          method => typeof module[method] === 'function'
+        )
+      ) {
+        const method = event.request.method ?? 'GET';
+        handler =
+          typeof module[method] === 'function'
+            ? module[method]
+            : () => new dependencies.Response(null, { status: 405 });
+      } else {
+        throw new Error(
+          'No default export was found. Add a default export to handle requests. Learn more: https://vercel.link/creating-edge-middleware'
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (non-blocking):** ‏add url to error

Can we add the url to the error message to make it easier to debug where this error comes from?

```suggestion
          `No default export was found at ${event.request.url}. Add a default export to handle requests. Learn more: https://vercel.link/creating-edge-middleware`
```